dns_server/src/crawler/mod.rs

```diff
@@ -0,0 +1,560 @@
+            }
+        }
+    }
+
+    fn handle_conn_response(
+        &mut self,
+        _peer_id: N::PeerId,
+        _request_id: N::PeerRequestId,
+        _response: PeerManagerResponse,
+    ) {
+        // Ignore all
+    }
+
+    fn handle_outbound_accepted(
+        &mut self,
+        address: N::Address,
+        peer_info: PeerInfo<N::PeerId>,
+        _receiver_address: Option<PeerAddress>,
+    ) {
+        let address_data = self
+            .addresses
+            .get_mut(&address)
+            .expect("address in the Connecting state must be known");
+        let valid = peer_info.network == self.config.network;
+
+        if !valid {
+            log::debug!("invalid peer detected at {}", address.to_string());
+            self.conn.disconnect(peer_info.peer_id).expect("disconnect must succeed");
+            Self::change_address_state(
+                &self.config,
+                &address,
+                address_data,
+                AddressState::Disconnected,
+                &mut self.storage,
+                &self.command_tx,
+            );
+            return;
+        }
+
+        Self::change_address_state(
+            &self.config,
+            &address,
+            address_data,
+            AddressState::Connected,
+            &mut self.storage,
+            &self.command_tx,
+        );
+        self.peers.insert(peer_info.peer_id, address);
+    }
+
+    fn handle_inbound_accepted(
+        &mut self,
+        _address: N::Address,
+        _peer_info: PeerInfo<N::PeerId>,
+        _receiver_address: Option<PeerAddress>,
+    ) {
+        unreachable!("unexpected inbound connection");
+    }
+
+    fn handle_connection_error(&mut self, address: N::Address, error: P2pError) {
+        log::debug!("connection to {} failed: {}", address.to_string(), error);
+        let address_data = self
+            .addresses
+            .get_mut(&address)
+            .expect("address in the Connecting state must be known");
+        Self::change_address_state(
+            &self.config,
+            &address,
+            address_data,
+            AddressState::Disconnected,
+            &mut self.storage,
+            &self.command_tx,
+        );
+    }
+
+    fn handle_connection_closed(&mut self, peer_id: N::PeerId) {
+        log::debug!("connection from peer {} closed", peer_id);
+        if let Some(address) = self.peers.remove(&peer_id) {
+            let address_data = self
+                .addresses
+                .get_mut(&address)
+                .expect("address in the Connected state must be known");
+            Self::change_address_state(
+                &self.config,
+                &address,
+                address_data,
+                AddressState::Disconnected,
+                &mut self.storage,
+                &self.command_tx,
+            );
+        }
+    }
+
+    fn handle_misbehaved(&mut self, _peer_id: N::PeerId, _error: P2pError) {
+        // Ignore all misbehave reports
```

Comment:
Are we sure? Why not disconnect them or remove them from lists?

Reply:
`NetworkingService` normally does not send `Misbehaved` messages (except for `ANNOUNCEMENT_MAX_SIZE` check), so I decided to keep things simpler.

Reply:
Don't you think this could open the door to flooding attacks or crash the node with huge messages?

Reply:
Huge messages are not a problem because the backend limits the maximum message size to 10MB.
Flooding attacks shouldn't be a problem either, because we don't do any processing.
Anyway, I added a TODO and created an issue (https://github.com/mintlayer/mintlayer-core/issues/699) for this.